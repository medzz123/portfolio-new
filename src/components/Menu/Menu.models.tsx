export interface MenuProps {
  label: string;
  items: {
    label: string;
    onClick: () => void;
  }[];
}
