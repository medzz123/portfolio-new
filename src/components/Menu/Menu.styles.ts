import { Content, Item, Trigger } from '@radix-ui/react-dropdown-menu';
import { styled } from '@theme/config';

export const StyledContent = styled(Content, {
  minWidth: '$3xs',
  backgroundColor: '$loContrast',
  borderRadius: '$sizes$6',
  padding: '$5',
  boxShadow: '$colors$shadow',
  mt: '$16',
});

export const StyledItem = styled(Item, {
  fontSize: '$info',
  color: '$hiContrast',
  padding: '$5 $10',
  borderRadius: '$sizes$3',
  cursor: 'default',

  '&:focus': {
    outline: 'none',
    backgroundColor: '$primary',
    color: '$loContrast',
  },
});

export const StyledTrigger = styled(Trigger, {
  fontSize: '$info',
  backgroundColor: 'transparent',
  color: '$hiContrast',
  border: 'none',
  boxShadow: 'none',
  padding: '$0',
  cursor: 'pointer',
  textDecoration: 'underline',
  display: 'flex',
  alignItems: 'center',
});
