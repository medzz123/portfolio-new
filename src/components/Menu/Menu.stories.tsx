import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import Menu from './Menu';

export default {
  title: 'Components/Menu',
  component: Menu,
} as Meta;

const Template: Story = (storyArguments) => (
  <Menu
    label="story"
    items={[
      {
        label: 'Item one',
        onClick: () => {
          // do nothing
        },
      },
    ]}
    {...storyArguments}
  />
);

export const Default = Template.bind({});
