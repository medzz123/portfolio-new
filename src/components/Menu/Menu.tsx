import { Root } from '@radix-ui/react-dropdown-menu';
import React, { FunctionComponent } from 'react';

import { MenuProps } from './Menu.models';
import { StyledContent, StyledItem, StyledTrigger } from './Menu.styles';

const Menu: FunctionComponent<MenuProps> = (props) => {
  const { label, items } = props;

  return (
    <Root>
      <StyledTrigger>{label} A</StyledTrigger>
      <StyledContent>
        {items.map((item) => (
          <StyledItem key={item.label} onSelect={item.onClick}>
            {item.label}
          </StyledItem>
        ))}
      </StyledContent>
    </Root>
  );
};

export default Menu;
