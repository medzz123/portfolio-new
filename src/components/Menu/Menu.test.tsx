import { render } from '@test/library';
import React from 'react';

import Menu from './Menu';

test('Menu Test', () => {
  const { container } = render(
    <Menu label="hello" items={[{ label: 'item-one', onClick: jest.fn }]} />
  );

  expect(container).toMatchSnapshot();
});
