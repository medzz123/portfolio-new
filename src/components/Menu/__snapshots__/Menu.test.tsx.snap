// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Test 1`] = `
<div>
  <button
    aria-haspopup="menu"
    class="sxqt3zu"
    data-state="closed"
    id="radix-id-0-1"
    type="button"
  >
    hello
     
    <i
      class="sx03kze sx03kze-8rfvp gg-chevron-down"
    />
  </button>
</div>
`;
